### DISCUSSION TITLE
Thermal Printer

### DISCUSSION THREAD

**Discussion by Tommiller0515** [selfcustody/krux#312] [Q&A]
Looking for info on getting a thermal printer connected to MAix Amigo krux. The link to Adafruit printer is discontinued but I see similar ones on AliExpress and eBay that I imagine would work. I see in krux printer settings rx/tx pin numbers but not sure which pins those numbers are associated with on MAix Amigo or what other printer connections are required. Any help would be greatly appreciated.

**Comment by tadeubas** [comment]
Hi, Maix Amigo already has this 3 ports on the bottom ready to receive a `Grove - 4 pin Male` connector, any thermal printer with TTL serial input may work. Plz use the bottom right connector when looking to the screen:

![sipeed_maix_amigo_vi](https://github.com/selfcustody/krux/assets/5518496/fa701282-a2a3-4807-93b5-3713cd2e5e1b)

**Reply by tadeubas** [reply]
As you see the **bottom right connector** has `ground, voltage, io7, io9` (from right to left looking to the screen). Plz **check if the Settings TX Pin is set to the value 7**
Now connect Amigo's `ground and TX` to printer `ground and RX`.

On this setup below: Grounds are black, Amigo's `TX` is white, printer `RX` is yellow
![image](https://github.com/selfcustody/krux/assets/5518496/a733eeb2-6638-4df9-86c2-d1513670215a)

**Reply by tadeubas** [reply]
Another Maix Amigo version

![photo2_2024-03-24_00-20-35](https://github.com/selfcustody/krux/assets/5518496/4bed4546-b218-4115-b473-c59c998e66fa)

**Comment by tadeubas** [comment]
For M5StickV there is a single grove connector, pins are `ground, voltage, io35, io34` (from right to left looking to the screen). Plz check the Settings `TX` Pin and use it to connect to the the printer `RX`. You will only need 2 cables, one for ground and other to send data to the printer.

![image](https://github.com/selfcustody/krux/assets/5518496/4773d39f-5f9d-4658-8b4e-4b3179c39c30)

**Comment by odudex** [comment]
Look for **Goojprt QR203** printer. We need to add more information about the setup, for now please note:
- You will have to solder custom cables.
- Printer comes with cables, but you'll need to cut and customize them. You'll need to add a GROVE male to one end to connect to your Amigo.
- You only need to connect GND and Amigo's Tx to printer's Rx.
- You need a dedicated power supply for the printer (5 to 9V, 3A), for which you also need a custom cable. Do NOT connect Amigo's 5V pin to your printer, as the printer demands more current Amigo can supply.

**Comment by tadeubas** [comment]
For Cube there is a single grove connector, pins are ground, voltage, io25, io24:

![image](https://github.com/selfcustody/krux/assets/5518496/fd6dc74e-823d-42b1-ba61-4ae0d9b242be)

![image](https://github.com/selfcustody/krux/assets/5518496/6e26bfe4-cf8d-4f4c-8170-9fca7a888d8e)

**Comment by tadeubas** [comment]
Yahboom comes with PH2.0 4Pin female connector, it is the same connector as this new scheme of the Goojprt Qr203 TTL printer.

Pins are Voltage, Ground, io8, io6

![yahboom3](https://github.com/selfcustody/krux/assets/5518496/e3d71357-a3f8-42bc-a857-9009a933e967)

![image](https://github.com/selfcustody/krux/assets/5518496/0ac2f2be-bdae-4b93-8398-2b4bf82f629a)

**Comment by tadeubas** [comment]
For Wonder MV, use the UART interface connector:

![image](https://github.com/user-attachments/assets/f48d7120-cfaf-4504-ad57-36d7cea6ae86)
